### PR TITLE
Deprecate interval collection conflict resolvers

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

Deprecates a no-longer relevant concept for future removal.

### CODE REVIEW

File: packages/dds/sequence/src/intervalCollection.ts [hunk 1]
```diff
@@ -1956,6 +1956,13 @@ export class IntervalCollection<TInterval extends ISerializableInterval> extends
...
 	}
 
+	/**
+	 * @deprecated - This functionality was useful when adding two intervals at the same start/end positions resulted
+	 * in a conflict. This is no longer the case (as of PR#6407), as interval collections support multiple intervals
+	 * at the same location and gives each interval a unique id.
+	 *
+	 * As such, the conflict resolver is never invoked and unnecessary. This API will be removed in an upcoming release.
+	 */
 	public addConflictResolver(conflictResolver: IntervalConflictResolver<TInterval>): void {
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/intervalCollection.ts:1966]: can you add a note to the breaking.md upcoming section, that's where we notify about deprecations that will be removed

Comment (Abe27342) [reply packages/dds/sequence/src/intervalCollection.ts:1966]: yeah good call. i've been doing it in next mostly with the removal (most breaks I've made are straightforward fixes) but makes more sense to do it with the deprecation. i'll do that here and in the future.

**Review by anthony-murphy — APPROVED** [review]

**Comment by Abe27342** [comment]
> @Abe27342 Mind if I add a changeset before you merge?

That's frustrating. Github UI updated *after* I hit the merge button on the tab I had open from before you commented. Sorry :/.

**Comment by tylerbutler** [comment]
> > @Abe27342 Mind if I add a changeset before you merge?
> 
> That's frustrating. Github UI updated _after_ I hit the merge button on the tab I had open from before you commented. Sorry :/.

No worries; here's a PR: #15089

**Pull request merged**